neurom/io/hdf5.py

```diff
@@ -184,29 +184,29 @@ def remove_duplicate_points(points, groups):
         '''
 
         group_initial_ids = groups[:, 0]
+        group_len = len(group_initial_ids)
 
         def _find_last_point(group_id):
             ''' Identifies and returns the id of the last point of a group'''
+            return group_initial_ids[group_id + 1] - 1
```

Comment:
I think there is a problem here:

group_initial_ids[group_id + 1] implies that the ordering between sections is correctly increasing. This is indeed True for the example I created and most of the trivial trees we generate. I would like to generate some more complicated trees to add to the testing. 

However, we should check if the sections are expected to be ordered:
So section_of_branch_order1 is directly followed by section_of_branch_order2. In this case, I think the modification is accurate.